vpr/src/place/net_cost_handler.cpp

```diff
@@ -0,0 +1,2277 @@
+                                               const PlaceDelayModel* delay_model,
+                                               const PlacerCriticalities* criticalities,
+                                               const ClusterBlockId& blk_id,
+                                               const ClusterPinId& pin_id,
+                                               const t_pl_moved_block& moving_blk_inf,
+                                               std::vector<ClusterPinId>& affected_pins,
+                                               double& timing_delta_c,
+                                               int& num_affected_nets,
+                                               bool is_src_moving);
+
+/**
+ * @brief Calculate the 3D bounding box of "net_id" from scratch (based on the block locations stored in place_ctx) and
+ * store them in bb_coord_new
+ * @param net_id
+ * @param bb_coord_new
+ * @param num_sink_pin_layer Store the number of sink pins of "net_id" on each layer
+ */
+static void get_non_updatable_bb(ClusterNetId net_id,
+                                  t_bb& bb_coord_new,
+                                  vtr::NdMatrixProxy<int, 1> num_sink_pin_layer);
+
+/**
+ * @brief Calculate the per-layer bounding box of "net_id" from scratch (based on the block locations stored in place_ctx) and
+ * store them in bb_coord_new
+ * @param net_id
+ * @param bb_coord_new
+ * @param num_sink_layer
+ */
+static void get_non_updatable_layer_bb(ClusterNetId net_id,
+                                       std::vector<t_2D_bb>& bb_coord_new,
+                                       vtr::NdMatrixProxy<int, 1> num_sink_layer);
+
+/**
+ * @brief Update the 3D bounding box of "net_id" incrementally based on the old and new locations of the pin
+ * @param bb_edge_new
+ * @param bb_coord_new
+ * @param num_sink_pin_layer_new
+ * @param pin_old_loc
+ * @param pin_new_loc
+ * @param src_pin
+ */
+static void update_bb(ClusterNetId net_id,
+                      t_bb& bb_edge_new,
+                      t_bb& bb_coord_new,
+                      vtr::NdMatrixProxy<int, 1> num_sink_pin_layer_new,
+                      t_physical_tile_loc pin_old_loc,
+                      t_physical_tile_loc pin_new_loc,
+                      bool src_pin);
+
+/**
+ * @brief Update the per-layer bounding box of "net_id" incrementally based on the old and new locations of the pin
+ * @param bb_edge_new
+ * @param bb_coord_new
+ * @param num_sink_pin_layer_new
+ * @param pin_old_loc
+ * @param pin_new_loc
+ * @param src_pin
+ */
+static void update_layer_bb(ClusterNetId net_id,
+                            std::vector<t_2D_bb>& bb_edge_new,
+                            std::vector<t_2D_bb>& bb_coord_new,
+                            vtr::NdMatrixProxy<int, 1> bb_pin_sink_count_new,
+                            t_physical_tile_loc pin_old_loc,
+                            t_physical_tile_loc pin_new_loc,
```

Comment:
Both `pin_new_loc` and `pin_old_loc` are used and modified in the function. I didn't pass the variables by reference because I wanted to avoid the overhead of creating new variables in the function. Instead, I am reusing the variables which are already pushed to the stack.

Reply:
OK.